### ISSUE TITLE
[autopilot] predict_proba columns follow first-appearance order, so sklearn metrics read them as the wrong classes

### ISSUE THREAD

**Issue by wdm0006** [wdm0006/petersburg#98]
## Problem

`classes_` is populated in **first-appearance order**, and `predict_proba`'s columns follow that
same order. scikit-learn's metrics assume a classifier's probability columns are ordered
**lexicographically** (the `np.unique` convention every sklearn estimator follows), so any caller
that hands `predict_proba` output to sklearn silently gets a number computed against the wrong
class-to-column mapping.

`_set_fitted_attributes` (`petersburg/estimators.py:192`) builds `classes_` directly from
`_terminal_indices(categories)`, i.e. the fitted `(layer, value)` list, which is first-appearance
ordered by design. Both `predict_proba` bodies then fill column `terminal.index(node_id)` —
`petersburg/estimators.py:377` (`FrequencyEstimator`) and `petersburg/estimators.py:622`
(`MixedModeEstimator`).

Measured on master (`2564090`), a fitted `FrequencyEstimator` on a two-layer target whose terminal
labels first appear as `win`, `draw`, `lose`:

```
classes_              : ['win' 'draw' 'lose']
np.unique(y[:, -1])   : ['draw' 'lose' 'win']
```

scikit-learn 1.7.2 says so itself, unprompted, on `log_loss(y_true, est.predict_proba(X))`:

```
UserWarning: Labels passed were [np.str_('win'), np.str_('draw'), np.str_('lose')].
But this function assumes labels are ordered lexicographically. Pass the ordered
labels=['draw', 'lose', 'win'] and ensure that the columns of y_prob correspond to
this ordering.
```

The damage is not cosmetic. On a skewed fixture (270 `win` / 20 `draw` / 10 `lose`,
`num_simulations=400`, `random_state=11`), where a predicted row is
`[0.9075, 0.065, 0.0275]` against `classes_ == ['win', 'draw', 'lose']`:

```
log_loss consumed as returned (sklearn assumes sorted) : 3.4504
log_loss with columns permuted to sorted order        : 0.1041
ratio                                                  : 33.1x
```

A confident, mostly-correct model reports as barely better than random. Nothing raises; only that
one `UserWarning` fires, and it does not fire at all for a caller that passes
`labels=list(est.classes_)` (sklearn re-sorts internally anyway) or for `roc_auc_score`.

Why the existing tests miss it: every `classes_` fixture in the suite happens to have
first-appearance order **coincide** with sorted order — `["x", "y"]` in
`TestFittedAttributes._data()` and in `test_fit_without_features_still_records_classes`,
`[0, 1]` in `test_classifier_backed_distributions_are_feature_dependent`. The one test with a
distinguishing fixture, `tests/test_estimators.py:888`
`test_classes_follow_first_appearance_order`, **asserts the current wrong order**
(`["late", "early"]`). And the sklearn check that would normally catch this,
`check_classifiers_classes`, is in the harness skip list (`tests/test_estimators.py:1172`)
because that harness fits a 1-D `y`.

`predict`, `score` and `_terminal_accuracy` are unaffected — they return and compare labels, never
column positions.

## Proposed change

Make `classes_` the sorted terminal labels, and map `predict_proba`'s columns through `classes_`
rather than through terminal-category position.

Confined to three places:

1. `_set_fitted_attributes` (`petersburg/estimators.py:192`) — derive `classes_` as the sorted
   unique terminal labels (`np.unique` over the terminal-layer values is the natural spelling;
   last-layer values are unique by construction, so this is a pure reorder, never a merge).
2. `FrequencyEstimator.predict_proba` (`petersburg/estimators.py:377`) — resolve a simulated
   terminal node id to its label, then to that label's index in `classes_`, instead of
   `terminal.index(node_id)`. Precompute the node-id-to-column map once per call so the inner loop
   stays O(1). Keep the existing "not a terminal category" `ValueError` for ids outside the final
   layer.
3. `MixedModeEstimator.predict_proba` (`petersburg/estimators.py:622`) — the identical change; the
   two bodies are duplicated verbatim today, so a shared module-level helper for the id-to-column
   map is welcome but optional.

Then invert `tests/test_estimators.py:888` `test_classes_follow_first_appearance_order` into a test
that `classes_` is sorted, renaming it accordingly.

**Explicitly out of scope: do not touch `_build_categories` or `_categories` ordering.** That list
is first-appearance ordered deliberately (it must not assume labels are mutually comparable, and
`TestCategoryOrderDeterminism` guards it). Only the *public* `classes_` attribute and the
`predict_proba` column mapping change; the internal category order, node ids, and `predict`'s
return value all stay exactly as they are.

## Acceptance

- [ ] `classes_` equals the sorted unique terminal labels for both estimator classes, verified on a
      fixture whose first-appearance order differs from sorted order (e.g. terminal labels appearing
      as `late`, `early`).
- [ ] `predict_proba` column `j` is the probability of `classes_[j]` — asserted by **per-column
      value**, on a fixture where the classes have distinctly different probabilities (the
      270/20/10 `win`/`draw`/`lose` shape above gives roughly `0.9 / 0.065 / 0.03`). A row-sum
      assertion or a near-balanced fixture passes on the broken code and does not count.
- [ ] `_build_categories` / `_categories` order is unchanged; `TestCategoryOrderDeterminism` passes
      untouched.
- [ ] `predict`, `score`, and `test_predict_matches_the_distribution_argmax_under_a_seed` still pass
      with no change to their assertions (the argmax test indexes through `classes_`, so it must
      stay green both before and after).
- [ ] Mutation check, reported in the PR body, run **per site**:
      (a) reverting `classes_` to first-appearance order fails the sorted-order assertion;
      (b) leaving the columns in terminal-index order while `classes_` is sorted fails the
      per-column value assertion;
      (c) reverting **only** `FrequencyEstimator.predict_proba` and **only**
      `MixedModeEstimator.predict_proba`, separately, each fails — fixing one site while leaving the
      other must not read as green.
- [ ] `log_loss(y_true, est.predict_proba(X))` on the skewed fixture emits no lexicographic-ordering
      `UserWarning` and returns the low value (~0.10), not the ~3.45 one.
- [ ] Full local gate passes: `ruff check`, `black --check`, `pytest --doctest-modules petersburg/`,
      `pytest` over `petersburg/ tests/ examples/`.

## Risk

**Medium.** This is a public behavior change to a surface added very recently: `classes_` ordering
and `predict_proba` column ordering both move for any fitted estimator whose terminal labels did not
happen to first appear in sorted order. That is the point of the fix — the new ordering is the one
sklearn consumers already assume — but a caller who hardcoded column indices against the old order
will read different numbers.

Two things bound the risk. Callers who index through `classes_` (the documented and tested pattern,
e.g. `list(est.classes_).index(1)` at `tests/test_estimators.py:843`) are correct before and after.
And `predict`/`score` are untouched, so the label-returning path — the one both bundled
`examples/estimation/` scripts use — cannot move. No in-repo example calls `predict_proba`.

Worth a CHANGELOG entry under `[Unreleased]`; note that open PR #97 is cutting 0.3.0, so this should
land before that tag or be recorded against the following version.
